libcni/api.go

```diff
@@ -165,33 +167,40 @@ func (c *CNIConfig) ensureExec() invoke.Exec {
 	return c.exec
 }
 
-func getResultCacheFilePath(netName string, rt *RuntimeConf) string {
+type cacheFileType string
+
+const (
+	cacheFileTypeResult cacheFileType = "results"
+	cacheFileTypeConfig cacheFileType = "config"
+)
+
+func getCacheFilePath(fileType cacheFileType, netName string, rt *RuntimeConf) string {
 	cacheDir := rt.CacheDir
 	if cacheDir == "" {
 		cacheDir = CacheDir
 	}
-	return filepath.Join(cacheDir, "results", fmt.Sprintf("%s-%s-%s", netName, rt.ContainerID, rt.IfName))
+	return filepath.Join(cacheDir, string(fileType), fmt.Sprintf("%s-%s-%s", netName, rt.ContainerID, rt.IfName))
```

Comment:
Need to escape all these variables - there's no guarantee that they are filename-safe.

Reply:
Ensuring these variables will be finename-safe will be address via https://github.com/containernetworking/cni/issues/679